src/main/kotlin/io/github/rybalkinsd/kohttp/dsl/context/UploadContext.kt

```diff
@@ -11,24 +11,68 @@ import java.net.URL
  * @author sergey
  */
 class UploadContext(private val context: HttpPostContext = HttpPostContext()) : IHttpContext by context {
+    /**
+     * Sets query parameters for File Upload.
+     * @param init Init body to define parameters
+     * @return Unit.
+     */
+    fun param(init: ParamContext.() -> Unit) {
```

Comment:
How do u think, is there a good way to delegate `param`, `header` and `url` to `HttpContext` or `HttpPostContext` ?

Reply:
I think one possible way to do that could be by leveraging `IHttpContext`, it currently has only `makeRequest`, but all its implementors have good number of overlapping functions. We could move them to `IHttpContext` enabling us to delegate through `HttpPostContext` on `UploadContext`. Thoughts?

Reply:
It's a right way in my mind, however, probably need to rethink interface name. And/or slit new methods to separate interfaces. Maybe even during several attempts. Basic interface(s) also will help us to do testing. I would also suggest, at least from the first version, not to make these interfaces public (internal would be better)